### PR TITLE
WIP - Support for .NET Core

### PR DESCRIPTION
**Pull request by maca88**

### CODE REVIEW

File: Source/AsyncGenerator/Internal/EnvironmentHelper.cs [hunk 1]
```diff
@@ -102,5 +99,43 @@ public static string GetNetCoreMsBuildPath()
...
+			// Linux -> file:///usr/lib/mono/4.5/mscorlib.dll
+			// OSX -> file:///Library/Frameworks/Mono.framework/Versions/5.2.0/lib/mono/4.5/mscorlib.dll
+			// MSBuild.dll is then located:
+			// Windows -> C:/Program Files (x86)/Mono/lib/mono/msbuild/15.0/bin/MSBuild.dll
+			// Linux -> /usr/lib/mono/msbuild/15.0/bin/MSBuild.dll
+			// OSX -> /Library/Frameworks/Mono.framework/Versions/5.2.0/lib/mono/msbuild/15.0/bin/MSBuild.dll
+
+			var assembly = typeof(System.Runtime.GCSettings).Assembly;
+			// Remove file:// on Unix or file:/// on Windows
+			var assemblyDirectory = Path.GetDirectoryName(assembly.CodeBase.Substring(IsWindows ? 8 : 7));
```
Comment (hazzik) [inline Source/AsyncGenerator/Internal/EnvironmentHelper.cs]: var assemblyDirectory = new Uri(assembly.CodeBase).LocalPath;

Comment (hazzik) [reply Source/AsyncGenerator/Internal/EnvironmentHelper.cs]: https://stackoverflow.com/questions/4107625/how-can-i-convert-assembly-codebase-into-a-filesystem-path-in-c

Comment (hazzik) [reply Source/AsyncGenerator/Internal/EnvironmentHelper.cs]: Also, please check this: https://blogs.msdn.microsoft.com/suzcook/2003/06/26/assembly-codebase-vs-assembly-location/

File: Source/AsyncGenerator.CommandLine/AsyncGenerator.CommandLine.csproj [hunk 2]
```diff
@@ -1,206 +1,58 @@
...
-    <DebugType>full</DebugType>
-    <Optimize>false</Optimize>
-    <OutputPath>bin\Debug\</OutputPath>
-    <DefineConstants>DEBUG;TRACE</DefineConstants>
-    <ErrorReport>prompt</ErrorReport>
-    <WarningLevel>4</WarningLevel>
-    <Prefer32Bit>false</Prefer32Bit>
+
+  <PropertyGroup Condition="'$(TargetFramework)' == 'net461'">
+    <DefineConstants>NET461</DefineConstants>
```
Comment (hazzik) [inline Source/AsyncGenerator.CommandLine/AsyncGenerator.CommandLine.csproj]: This should be default

File: Source/AsyncGenerator.CommandLine/AsyncGenerator.CommandLine.csproj [hunk 3]
```diff
@@ -1,206 +1,58 @@
...
-    <PlatformTarget>AnyCPU</PlatformTarget>
-    <DebugType>pdbonly</DebugType>
-    <Optimize>true</Optimize>
-    <OutputPath>bin\Release\</OutputPath>
-    <DefineConstants>TRACE</DefineConstants>
-    <ErrorReport>prompt</ErrorReport>
-    <WarningLevel>4</WarningLevel>
-    <Prefer32Bit>false</Prefer32Bit>
+  <PropertyGroup Condition="'$(TargetFramework)' == 'netcoreapp2.0'">
+    <DefineConstants>NETCORE2</DefineConstants>
```
Comment (hazzik) [inline Source/AsyncGenerator.CommandLine/AsyncGenerator.CommandLine.csproj]: Same here - it's by default.

File: Source/AsyncGenerator/Internal/EnvironmentHelper.cs (hunk 1 above)
Comment (maca88) [reply Source/AsyncGenerator/Internal/EnvironmentHelper.cs]: :+1:

**Comment by maca88** [comment]
Replaced by #125

**Pull request closed without merging**